miden-tx/src/tests.rs

```diff
@@ -306,3 +310,146 @@ fn test_prove_and_verify_with_tx_executor() {
+
+#[test]
+fn test_p2id_script() {
+    // MOCK DATA
+    // --------------------------------------------------------------------------------------------
+    let mut assembler = assembler();
+
+    // Create assets and sender account ID
+    // Create Note and all assets
+    let faucet_id_1 = AccountId::try_from(ACCOUNT_ID_FUNGIBLE_FAUCET_ON_CHAIN).unwrap();
+    let fungible_asset_1: Asset = FungibleAsset::new(faucet_id_1, 100).unwrap().into();
+    let sender_account_id = AccountId::try_from(ACCOUNT_ID_SENDER).unwrap();
+
+    // Create the target account that receives the note
+    let target_account_id =
+        AccountId::try_from(ACCOUNT_ID_REGULAR_ACCOUNT_UPDATABLE_CODE_ON_CHAIN).unwrap();
+
+    // TODO: We don't have the add_asset procedure in the assembler yet, we don't need custom code
+    const TARGET_ACCOUNT_CODE_MASM: &'static str = "\
+    export.account_proc_1
+        push.9.9.9.9
+        dropw
+    end
+    ";
+    let target_account_code_ast = ModuleAst::parse(TARGET_ACCOUNT_CODE_MASM).unwrap();
+    let target_account_code =
+        AccountCode::new(target_account_id, target_account_code_ast, &mut assembler).unwrap();
+    let account = Account::new(
+        target_account_id,
+        AccountVault::default(),
+        AccountStorage::default(),
+        target_account_code.clone(),
+        ONE,
+    );
+
+    // Create the note with the P2ID script w/ one asset
+    let note_program_ast =
+        ProgramAst::parse(
+            format!(
+                "
+                use.miden::sat::account
+                use.miden::sat::note
+                
+                begin                                                   # [note_inputs = target_account_id, ...]                                            
+                    exec.account::get_id                                # [account_id, target_account_id, ...]
+                    eq                                                  # [account_id == target_account_id, ...]
+                    assert                                              # [] if account_id == target_account_id, fails if not
+
+                    push.1000000000                                     # [1000000000, ...] memory pointer to store assets
+                    exec.note::get_assets                               # [num_of_assets, 1000000000, ...]    
+                    
+                    dup push.0 gt                                       # [1 || 0, num_of_assets, 1000000000, ...]
+                    while.true                                          # [num_of_assets, 1000000000, ...]
+                        exec.account::get_nonce drop                    # TODO: Should call add_asset but we don't have it yet
+                        sub.1                                           # [num_of_assets - 1, 1000000000, ...] u32checked_sub not needed
+                        push.0 gt                                       # [1, ...], if num_of_assets - 1 > 0, [0, ...] otherwise
```

Comment:
same as above - maybe we can change to use `neq` / `eq` instead.

Reply:
Very good point